estraverse.js

```diff
@@ -432,7 +432,12 @@
         this.__leavelist = [];
         this.__current = null;
         this.__state = null;
-        this.__fallback = visitor.fallback === 'iteration';
+        this.__fallback = visitor.fallback === 'iteration' || typeof visitor.fallback === 'function';
+        this.__fallbackFunction = objectKeys;
```

Comment:
I don't think we need two pieces of state here.

```js
this.__fallback = null;
if (visitor.fallback === 'iteration') this.__fallback = objectKeys;
else if (typeof visitor.fallback === 'function') this.__fallback = visitor.fallback;
```

Then just check that it's non-null before calling it below.